src/core/public/chrome/ui/header/header_logo.tsx

```diff
@@ -54,47 +55,47 @@ function onClick(
     return;
   }
 
-  if (
-    !forceNavigation ||
-    event.isDefaultPrevented() ||
-    event.altKey ||
-    event.metaKey ||
-    event.ctrlKey
-  ) {
+  if (event.isDefaultPrevented() || event.altKey || event.metaKey || event.ctrlKey) {
     return;
   }
 
-  const toParsed = Url.parse(anchor.href);
-  const fromParsed = Url.parse(document.location.href);
-  const sameProto = toParsed.protocol === fromParsed.protocol;
-  const sameHost = toParsed.host === fromParsed.host;
-  const samePath = toParsed.path === fromParsed.path;
+  if (forceNavigation) {
+    const toParsed = Url.parse(anchor.href);
+    const fromParsed = Url.parse(document.location.href);
+    const sameProto = toParsed.protocol === fromParsed.protocol;
+    const sameHost = toParsed.host === fromParsed.host;
+    const samePath = toParsed.path === fromParsed.path;
 
-  if (sameProto && sameHost && samePath) {
-    if (toParsed.hash) {
-      document.location.reload();
-    }
+    if (sameProto && sameHost && samePath) {
+      if (toParsed.hash) {
+        document.location.reload();
+      }
 
-    // event.preventDefault() keeps the browser from seeing the new url as an update
-    // and even setting window.location does not mimic that behavior, so instead
-    // we use stopPropagation() to prevent angular from seeing the click and
-    // starting a digest cycle/attempting to handle it in the router.
-    event.stopPropagation();
+      // event.preventDefault() keeps the browser from seeing the new url as an update
+      // and even setting window.location does not mimic that behavior, so instead
+      // we use stopPropagation() to prevent angular from seeing the click and
+      // starting a digest cycle/attempting to handle it in the router.
+      event.stopPropagation();
+    }
+  } else {
+    navigateToApp('home');
+    event.preventDefault();
```

Comment:
The (probably over-engineered and unused) logic in the `onClick` / `findClosestAnchor` methods was to find the navlink associated this the `HeaderLogo`'s url to navigate to it. As you replaced this with a simple `navigateToApp('home')`, maybe the logic using list of navlinks can be totally removed? We could just pass the info that is `home` navlink/app is, or not, disabled instead?